File: Data Structures & Algorithms/house-robber-ii/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[n-1]
        return max(self.f(nums[1:]), self.f(nums[:n-1]))

    def f(self, nums: List[int]) -> int:
        minusTwo, minusOne = 0, 0
        for n in nums:
            cur = max(minusTwo + n, minusOne)
            minusTwo = minusOne
            minusOne = cur
        return minusOne
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const set = Set(nums)
        let longest = 0;
        for(const n of nums) {
            if(!(set.has(n-1))) {
                let length = 0;
                while(set.has(n + length)) {
                    length += 1;
                }
                longest = Math.max(longest, length)
            } 
        }
        return longest;
    }
}
